hr_expense_vendor_receipt/models/hr_expense_sheet.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2025 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import models
+
+
+class HrExpenseSheet(models.Model):
+    _inherit = "hr.expense.sheet"
+
+    def _prepare_bills_vals(self):
+        result = super()._prepare_bills_vals()
+        if self.journal_id.receipts:
+            result.update({"move_type": "in_receipt"})
+        return result
+
+    def _do_reverse_moves(self):
+        moves_receipts_sudo = self.sudo().account_move_ids.filtered(
+            lambda move: move.state != "draft" and move.journal_id.receipts
+        )
+        moves_receipts_sudo.button_draft()
+        self.account_move_ids = self.account_move_ids - moves_receipts_sudo
+        return super()._do_reverse_moves()
```

Comment:
If you are going to set to draft, you must target "posted" moves. 
Use filtered_domain for faster iterations.

I don't like the idea to clear from expense sheets the receipts.


```suggestion
        moves_receipts_sudo = self.sudo().account_move_ids.filtered_domain([
            ("state", "=", "posted"),
            ("journal_id.receipts", "=", True),
        ])
        moves_receipts_sudo.button_draft()
        moves_receipts_sudo.button_cancel()
        # These draft recepits will be deleted when "_do_reverse_moves" happen
        return super()._do_reverse_moves()
```